core/include/detray/coordinates/line2.hpp

```diff
@@ -112,6 +112,20 @@ struct line2 : public coordinate_base<line2, transform3_t> {
         return locZ_in_global + p[0] * vector::normalize(r);
     }
 
+    /// @returns the normal vector
+    template <typename mask_t>
+    DETRAY_HOST_DEVICE inline vector3 normal(const transform3_t &trf3,
```

Comment:
Hmm. Practically we don't need normal function for line surface, do we? and it seems the return value (`trf3.z()`) doesn't sound right. 

I think you added this because of normal function in `surface_kernels` and `surface` but do we need to define it? or Is it supposed to be used by user?. 

I would not add `normal` function to `line2` and `surface` at all, or define the return value differently, e.g. direction of track

Reply:
I need some sort of surface normal to calculate the incidence angle. For the line surface I took what I found in ACTS (which does not mean its correct) and also how we calculated the angle in the line intersector. I did not want to change the results we get for the incidence angles for the lines, to make sure that #505 produces unchanged results (it is based on this PR). Can we correct the line surface normal in a different PR?

Reply:
I think returning trf.z() seems OK for `line2` after looking into `line_intersector.hpp`.
Could you check out the incidence angle calculation of other intersectors and use `cos_angle` instead

Reply:
The angle calculation in the intersectors is not the same, because it can make use of intermediate results that are already obtained elsewhere in the respective intersector. Also, the surface class would do the tuple unrolling all over again, which seems unnecessary at that stage. 
I only left the angle as part of the intersection, because it can be calculated more efficiently in the intersectors than in the surface. If we want to switch to the ```cos_angle``` implementation of the surface everywhere, then we should not do the angle calculation in the intersectors imo

Reply:
Sorry, what I meant to say is, the math is the same, but the way the calculation is done is more efficient in the intersectors right now imo

Reply:
I see. I think I get your point - let's think about better implementation later

Reply:
Ok